cve_bin_tool/util.py

```diff
@@ -3,7 +3,71 @@
 import fnmatch
 import os
 import sys
+from collections import defaultdict
 from enum import Enum
+from typing import NamedTuple
+
+
+class OrderedEnum(Enum):
+    def __ge__(self, other):
+        if self.__class__ is other.__class__:
+            return self.value >= other.value
+        return NotImplemented
+
+    def __gt__(self, other):
+        if self.__class__ is other.__class__:
+            return self.value > other.value
+        return NotImplemented
+
+    def __le__(self, other):
+        if self.__class__ is other.__class__:
+            return self.value <= other.value
+        return NotImplemented
+
+    def __lt__(self, other):
+        if self.__class__ is other.__class__:
+            return self.value < other.value
+        return NotImplemented
+
+
+class Remarks(OrderedEnum):
+    NewFound = 1, "1", "NewFound", "n", "N"
+    Unexplored = 2, "2", "Unexplored", "u", "U", ""
+    Confirmed = 3, "3", "Confirmed", "c", "C"
+    Mitigated = 4, "4", "Mitigated", "m", "M"
+    Ignored = 5, "5", "Ignored", "i", "I"
+
+    def __new__(cls, value, *aliases):
+        obj = object.__new__(cls)
+        obj._value_ = value
+        for alias in aliases:
+            cls._value2member_map_[alias] = obj
+        return obj
+
+
+class CVE(NamedTuple):
+    cve_number: str
+    severity: str
+    remarks: Remarks = Remarks.NewFound
+    description: str = ""
+    comments: str = ""
+
+
+class ProductInfo(NamedTuple):
+    vendor: str
+    product: str
+    version: str
+
+
+class CVEData(defaultdict):
+    def __missing__(self, key):
+        if key == "cves":
+            self[key] = []
+        elif key == "paths":
+            self[key] = set()
+        else:
+            return NotImplemented
+        return self[key]
```

Comment:
Issues with old structure:
1. Old CVEData was NamedTuple and since newly added path attribute was mutable it can create hard to find bugs. 
2. To update path we need to scan all_cve_data to find product for which we want to append paths.             
 `Time Complexity: O(n**2)` which can be reduced to `O(n)` using better structure.
3. Throwing vendor, product, version in different function was decreasing readability. So, ProductInfo would be nice to pack this data together since we don't actually need that alone. 
4. TriageData structure wasn't syncing with old CVEData. So, csv2cve was breaking.

New structure is addressing all these issues.